tests/integration/ambient/bookinfo_test.go

```diff
@@ -107,8 +107,32 @@ func TestBookinfo(t *testing.T) {
 				})
 			})
 
+			t.NewSubTest("ingress receives waypoint updates").Run(func(t framework.TestContext) {
+				setupWaypoints(t, nsConfig, "bookinfo-productpage")
+				retry.UntilSuccessOrFail(t, func() error {
```

Comment:
A bit worried about `Get; Delete; Get` in the retry loop? if it retries, it will already be delete

Reply:
are you suggesting waypoint-add and waypoint-delete should both be in the `retry.UntilSuccessOrFail` block?  I know successive istioctl waypoint-add commands do not throw any errors if the waypoint already exists so it's reasonable in that regard

Reply:
I think we may want

```
add
retry { get }
delete
retry {get }
```

maybe?

add+delete in the loop may not be great. I had thought delete 2x would fail, but maybe its already safe as is

Reply:
istioctl unfortunately throws an error on the delete if it can't find the gateway in question - was wondering if we needed to change the logic on that

Reply:
so does `kubectl delete` fwiw... so it may be appropriate

Reply:
maybe a WARN for istioctl instead of error on deletes?